store/tikv/coprocessor.go

```diff
@@ -1442,49 +1414,25 @@ func (e *rateLimitAction) destroyTokenIfNeeded(returnToken func()) {
 	if !e.cond.isTokenDestroyed {
 		e.cond.remainingTokenNum = e.cond.remainingTokenNum - 1
 		e.cond.isTokenDestroyed = true
-		e.cond.Broadcast()
+		e.cond.once = sync.Once{}
 		return
 	}
-
 	returnToken()
-	// we suspend worker when `exceeded` is true until being notified by `broadcastIfNeeded`
-	for e.cond.exceeded {
-		e.cond.waitingWorkerCnt++
-		e.cond.Wait()
-		e.cond.waitingWorkerCnt--
-	}
-	e.unsafeInitOnce()
-}
-
-// unsafeInitOnce would init once if the condition is meet. This should be used under condition's lock.
-func (e *rateLimitAction) unsafeInitOnce() {
-	// only when all the waiting workers have been resumed, the Action could be initialized again.
-	if e.cond.waitingWorkerCnt < 1 && e.cond.triggered {
-		e.cond.triggered = false
-		e.cond.once = sync.Once{}
-	}
 }
 
 func (e *rateLimitAction) conditionLock() {
-	e.cond.L.Lock()
+	e.cond.Lock()
 }
 
 func (e *rateLimitAction) conditionUnlock() {
-	e.cond.L.Unlock()
+	e.cond.Unlock()
 }
 
 func (e *rateLimitAction) close() {
 	if !e.isEnabled() {
 		return
 	}
 	e.setEnabled(false)
-	e.conditionLock()
-	defer e.conditionUnlock()
-	e.cond.exceeded = false
```

Comment:
when will `exceeded` be set to false after this modification?

Reply:
IMO, I think it's unnecessary to set `exceed` as false here, but It's ok to reset `exceed` in `close` function, so I add it back.